src/agentscope/models/dashscope_model.py

```diff
@@ -542,3 +510,319 @@ def __call__(
+                    ]
+                },
+                {
+                    "role": "user",
+                    "content": [
+                        {
+                            "text": (
+                                "## Dialogue History\n"
+                                "Bob: How about this picture?\n"
+                                "user: It's wonderful! How about mine?"
+                            )
+                        },
+                        {"image": "figure2"},
+                        {"image": "figure3"},
+                    ]
+                }
+            ]
+
+        Args:
+            args (`Union[MessageBase, Sequence[MessageBase]]`):
+                The input arguments to be formatted, where each argument
+                should be a `Msg` object, or a list of `Msg` objects.
+                In distribution, placeholder is also allowed.
+
+        Returns:
+            `List[dict]`:
+                The formatted messages.
+        """
+
+        # Parse all information into a list of messages
+        input_msgs = []
+        for _ in args:
+            if isinstance(_, MessageBase):
+                input_msgs.append(_)
+            elif isinstance(_, list) and all(
+                isinstance(__, MessageBase) for __ in _
+            ):
+                input_msgs.extend(_)
+            else:
+                raise TypeError(
+                    f"The input should be a Msg object or a list "
+                    f"of Msg objects, got {type(_)}.",
+                )
+
+        messages = []
+
+        # record dialog history as a list of strings
+        dialogue = []
+        image_or_audio_dicts = []
+        for i, unit in enumerate(input_msgs):
+            if i == 0 and unit.role == "system":
+                # system prompt
+                content = [{"text": _convert_to_str(unit.content)}]
+                content.extend(self._convert_url(unit.url))
+
+                messages.append(
+                    {
+                        "role": unit.role,
+                        "content": content,
+                    },
+                )
+            else:
+                # text message
+                dialogue.append(
+                    f"{unit.name}: {_convert_to_str(unit.content)}",
+                )
+                # image and audio
+                image_or_audio_dicts.extend(self._convert_url(unit.url))
+
+        dialogue_history = "\n".join(dialogue)
+
+        user_content_template = "## Dialogue History\n{dialogue_history}"
+
+        messages.append(
+            {
+                "role": "user",
+                "content": [
+                    {
+                        "text": user_content_template.format(
+                            dialogue_history=dialogue_history,
+                        ),
+                    },
+                    *image_or_audio_dicts,
+                ],
+            },
+        )
+
+        return messages
```

Comment:
In this case, the `image_or_audio_dicts` contains all the history images, I wonder if it will work or not, and I'm unsure if the output of the model is what the user wants.

Reply:
I have tested the following cases.
```python
import dashscope

dashscope.api_key="xxx"

res = dashscope.MultiModalConversation.call(
    model="qwen-vl-plus",
    messages=[
        {
            "role": "user",
            "content": [
                {
                    "text": "What's the difference between these two figures? "
                },
                {
                    "image": "https://img.alicdn.com/imgextra/i4/O1CN01kUiDtl1HVxN6G56vN_!!6000000000764-2-tps-43-19.png",
                },
                {
                    "image": "https://dashscope.oss-cn-beijing.aliyuncs.com/images/dog_and_girl.jpeg",
                }
            ]
        }
    ]
)

print(res)
```
The execution results:
```bash
{"status_code": 200, "request_id": "62449625-770c-9a56-954e-3ba5725f941f", "code": "", "message": "", "output": {"text": null, "finish_reason": null, "choices": [{"finish_reason": "stop", "message": {"role": "assistant", "content": [{"text": "The first figure is an orange \"NEW\" sign, while the second image shows a woman and her dog on the beach at sunset. They seem to be enjoying each other's company in this serene setting. The main differences lie in their nature - one being a simple text symbol representing newness or freshness (the 'new' sign), whereas the other depicts a living interaction with animals and humans sharing time together outdoors during beautiful natural scenery (woman & dog)."}]}}]}, "usage": {"input_tokens": 1300, "output_tokens": 92, "image_tokens": 1255}}
```

It seems like placing two images in a single message is okay for multimodal conversation api.